lib/ui/lerp.dart

```diff
@@ -26,7 +26,7 @@ double? lerpDouble(num? a, num? b, double t) {
 ///
 /// Same as [lerpDouble] but specialized for non-null `double` type.
 double _lerpDouble(double a, double b, double t) {
-  return a * (1.0 - t) + b * t;
+  return a + (b - a) * t;
```

Comment:
I think this is what may be causing the breakages. Throw inf in here for a or b and some lerp cases now produce nans:

![image](https://github.com/user-attachments/assets/c71b308b-7a7a-4d4b-8753-249575af2872)